platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptor.java

```diff
@@ -155,276 +111,129 @@ public GuestInterceptor(SecurityManager securityManager,
+            return;
+        }
 
-            //Check if security header exists; if not, execute GuestInterceptor logic
-            String actor = (String) getOption(WSHandlerConstants.ACTOR);
-            if (actor == null) {
-                actor = (String) message.getContextualProperty(SecurityConstants.ACTOR);
+        SOAPMessage soapMessage = getSOAPMessage(message);
+        internalHandleMessage(message, soapMessage);
+        if (LOGGER.isTraceEnabled()) {
+            try {
+                LOGGER.trace("SOAP request after guest interceptor: {}",
+                        XMLUtils.prettyFormat(soapMessage.getSOAPHeader()
+                                .getParentNode()));
+            } catch (SOAPException e) {
+                //ignore
             }
+        }
+
+    }
+
+    private void internalHandleMessage(SoapMessage message, SOAPMessage soapMessage) throws Fault {
+
+        //Check if security header exists; if not, execute GuestInterceptor logic
+        String actor = (String) getOption(WSHandlerConstants.ACTOR);
+        if (actor == null) {
+            actor = (String) message.getContextualProperty(SecurityConstants.ACTOR);
+        }
+
+        Element existingSecurityHeader = null;
+        try {
+            LOGGER.debug("Checking for security header.");
+            existingSecurityHeader = WSSecurityUtil.getSecurityHeader(soapMessage.getSOAPPart(),
+                    actor);
+        } catch (WSSecurityException e1) {
+            LOGGER.debug("Issue with getting security header", e1);
+        }
+
+        if (existingSecurityHeader != null) {
+            LOGGER.debug(
+                    "SOAP message contains security header, no action taken by the GuestInterceptor.");
+            return;
+        }
+
+        LOGGER.debug("Current request has no security header, continuing with GuestInterceptor");
+
+        AssertionInfoMap assertionInfoMap = message.get(AssertionInfoMap.class);
 
-            Element existingSecurityHeader = null;
+        EffectivePolicy effectivePolicy = getEffectivePolicy(message);
+
+        //Auto analyze endpoint policies
+
+        // if there is a policy, try to follow it as closely as possible
+        if (effectivePolicy == null) {
+            LOGGER.warn("Effective WS Policy is null no guest assertion will be generated.");
+            return;
+        }
+
+        Policy policy = effectivePolicy.getPolicy();
+        if (policy == null) {
+            LOGGER.warn("WS Policy is null no guest assertion will be generated.");
+            return;
+        }
+
+        assertionInfoMap.entrySet()
+                .stream()
+                .flatMap(p -> p.getValue()
+                        .stream())
+                .filter(info -> MetadataConstants.ADDRESSING_ASSERTION_QNAME.equals(info.getAssertion()
+                        .getName()))
+                .findFirst()
+                .ifPresent(tru -> createAddressing(message, soapMessage));
```

Comment:
I'm not a fan of the side-effect use of `ifPresent`. If you're not going to need the filtered object from the optional, I'd as soon see a terminating call to `isPresent` instead to get a boolean. Something like...
```
boolean hasAddressingAssertion = ...
    .isPresent();
if (hasAddressingAssertion) {
    createAddressing(message, soapMessage);
}
```